src/Ng/CommandHelpers.cs

```diff
@@ -363,5 +374,52 @@ public static Func<HttpMessageHandler> GetHttpMessageHandlerFactory(bool verbose
 
             return handlerFunc;
         }
+
+        public static IEnumerable<EndpointFactory.Input> GetEndpointFactoryInputs(IDictionary<string, string> arguments)
+        {
+            return arguments.GetOrThrow<string>(Arguments.EndpointsToTest).Split(';').Select(e => {
+                var endpointParts = e.Split('|');
+
+                if (endpointParts.Count() < 2)
+                {
+                    throw new ArgumentException(
+                        $"\"{e}\" is not a valid endpoint! Each endpoint must follow this format: " +
+                        $"<endpoint-to-test-name>|<endpoint-to-test-cursor>.");
+                }
+
+                return new EndpointFactory.Input(endpointParts[0], new Uri(endpointParts[1]));
+            });
+        }
+
+        public static IPackageMonitoringStatusService GetPackageMonitoringStatusService(IDictionary<string, string> arguments, CatalogIStorageFactory storageFactory, ILoggerFactory loggerFactory)
+        {
+            return new PackageMonitoringStatusService(
+                new NamedStorageFactory(storageFactory, arguments.GetOrDefault(Arguments.PackageStatusFolder, Arguments.PackageStatusFolderDefault)),
+                loggerFactory.CreateLogger<PackageMonitoringStatusService>());
+        }
+
+        public static IStorageQueue<T> CreateStorageQueue<T>(IDictionary<string, string> arguments)
+        {
+            var storageType = arguments.GetOrThrow<string>(Arguments.StorageType);
+
+            if (storageType.Equals(Arguments.FileStorageType, StringComparison.InvariantCultureIgnoreCase))
+            {
+                throw new NotImplementedException("Only Azure storage queues are supported!");
+            }
+
+            if (Arguments.AzureStorageType.Equals(storageType, StringComparison.InvariantCultureIgnoreCase))
+            {
+                var storageAccountName = arguments.GetOrThrow<string>(Arguments.StorageAccountName);
+                var storageKeyValue = arguments.GetOrThrow<string>(Arguments.StorageKeyValue);
+                var storageQueueName = arguments.GetOrDefault<string>(Arguments.StorageQueueName);
+
+                var credentials = new StorageCredentials(storageAccountName, storageKeyValue);
+                var account = new CloudStorageAccount(credentials, true);
+                return new AzureStorageQueue<T>(account, storageQueueName,
+                    new JsonStorageQueueMessageSerializer<T>(JsonSerializerUtility.SerializerSettings));
+            }
+
+            throw new ArgumentException($"Unrecognized storageType \"{storageType}\"");
```

Comment:
nit: simply couple this exception with the `NotImplementedException` above and throw storageType is not a `AzureStorageType`.

Reply:
good catch